apps/mobile/src/components/widgets/collectibles/collectibles.layout.tsx

```diff
@@ -0,0 +1,81 @@
+import { NonFungibleCryptoAssetInfo } from '@leather.io/models';
+import { Box, CollectibleCard } from '@leather.io/ui/native';
+import { isDefined, isEmptyString } from '@leather.io/utils';
+
+import { CollectibleCaption } from './collectible-caption';
+
+interface CollectiblesLayoutProps {
+  collectibles: NonFungibleCryptoAssetInfo[];
+  view?: 'widget' | 'gallery';
+}
+
+function filterCollectibles(collectible: NonFungibleCryptoAssetInfo) {
+  switch (collectible.protocol) {
+    case 'inscription':
+      return !isEmptyString(collectible.src) && isDefined(collectible.src);
+    case 'sip9':
+      return (
+        !isEmptyString(collectible.cachedImage) &&
+        isDefined(collectible.cachedImage) &&
+        // BNS - Archive is not a valid collectible
+        collectible.name !== 'BNS - Archive'
+      );
+    default:
+      return false;
+  }
+}
+
+export function CollectiblesLayout({ collectibles, view = 'gallery' }: CollectiblesLayoutProps) {
+  const displayLimit = 9;
+  const thumbnailSize = view === 'widget' ? 200 : 164;
+  return (
+    <Box flexDirection="row" alignItems="center" paddingHorizontal="5" gap="4" flexWrap="wrap">
+      {collectibles.filter(filterCollectibles).map((collectible, index) => {
+        if (view === 'widget' && index >= displayLimit) return null;
+        switch (collectible.protocol) {
+          case 'inscription':
+            return (
+              <CollectibleCard
+                key={index}
+                name={collectible.title}
+                type={collectible.protocol}
+                src={collectible.src}
+                mimeType={collectible.mimeType}
+                size={thumbnailSize}
+                caption={
+                  view === 'gallery' && (
+                    <CollectibleCaption
```

Comment:
I've improvised adding this caption to `gallery` view, copying the extension.  

It's not in the design but it's helpful for now, even just for debugging. Looking at the data and we seem to have some duplicate collectibles so more filtering could be needed.

Displaying the name makes that easier for now